### PR TITLE
ONI-185: Added validation for urls in attachment.

### PR DESCRIPTION
**Pull request by wzglinieckisoldevelo**
Ticket: https://openimis.atlassian.net/browse/ONI-185

FE PR: https://github.com/openimis/openimis-fe-claim_js/pull/176

Changes:
- Fixed typo in claim pdf rendering
- Added url validation in claim attachment on setting
- Setting let you specify allowed domains

### CODE REVIEW

File: claim/apps.py [hunk 1]
```diff
@@ -31,7 +31,8 @@
     "claim_validation_multiple_services_explanation_required": True,
     "autogenerate_func": 'claim.utils.autogenerate_nepali_claim_code',
     "additional_diagnosis_number_allowed": 4,
-    "claim_max_restore": None
+    "claim_max_restore": None,
+    "allowed_domains_attachments": ["claimdoc.hib.gov.np"]
```
Comment (dragos-dobre) [inline claim/apps.py:35]: Should be empty by default.

File: claim/gql_mutations.py [hunk 2]
```diff
@@ -268,9 +270,19 @@ def create_attachment(claim_id, data):
...
     from core import datetime
     now = datetime.datetime.now()
-    if ClaimConfig.claim_attachments_root_path:
-        # don't use data date as it may be updated by user afterwards!
-        data['url'] = create_file(now, claim_id, data.pop('document'))
+    general_type = data['general_type']
+    data['module'] = 'claim'
+    if general_type == GeneralClaimAttachmentType.URL:
+        parsed_url = urlparse(data['url'])
+        if (ClaimConfig.allowed_domains_attachments and
```
Comment (dragos-dobre) [inline claim/gql_mutations.py:277]: If config there but empty, is this checked?

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: claim/gql_mutations.py (hunk 2 above)
Comment (wzglinieckisoldevelo) [reply claim/gql_mutations.py:277]: ClaimConfig.allowed_domains_attachments defaults to false if list is empty, so whole expression is false.

File: claim/apps.py (hunk 1 above)
Comment (wzglinieckisoldevelo) [reply claim/apps.py]: Done

**Review by delcroip — APPROVED** [review]

**Pull request merged**